Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -449,9 +449,19 @@ bool EventHandler::handle_mousedown(CSSPixelPoint viewport_position, CSSPixelPoi
                         HTML::run_unfocusing_steps(focused_element);
                 }
 
-                // If we didn't focus anything, place the document text cursor at the mouse position.
-                // FIXME: This is all rather strange. Find a better solution.
-                if (!did_focus_something || dom_node->is_editable()) {
+                // Ask the next non-shadow parent element whether the node at the mouse position is selectable.
+                auto& root_node = dom_node->root();
+                DOM::Element* non_shadow_parent_element;
+                if (root_node.is_shadow_root())
+                    non_shadow_parent_element = root_node.parent_or_shadow_host_element();
+                else
+                    non_shadow_parent_element = dom_node->parent_element();
+                bool is_selectable = true;
+                if (non_shadow_parent_element && non_shadow_parent_element->is_html_element())
+                    is_selectable = static_cast<HTML::HTMLElement*>(non_shadow_parent_element)->is_child_node_selectable(dom_node.ptr());
```

Comment:
We know from line 434 that `dom_node` can't be null here. Making `is_child_node_selectable()` take a `DOM::Node const&` instead would be a little safer. (And make more sense IMO, because nobody should be asking if *nothing* is selectable.)